File: issta2018artifacts/minimized-bugs/buggy-clousot/Daikon/FourElementArray.cs
using System;
using System.Collections.Generic;
using System.Diagnostics.Contracts;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

class FourElementArray
{
    /* Every null in the function below was originally some meaningful String.
     * The original documentation specified that this function always returns
     * an array with four elements.
     */
    public static string[] esc_quantify(params object[] vars)
    {
        Contract.Ensures(Contract.Result<string[]>().Length >= 4);

        if (vars.Length == 1)
        {
            //:: error: return.type.incompatible
            return new string[] { null, null, ")" };
        }
        else if ((vars.Length == 2) && boolean_condition(vars[1]))
        {
            return new string[] {
                null,
                null,
                null,
                ")"
            };
        }
        else
        {
            return new string[] {
                null, null, null, ")"
            };
        }
    }

    /* In the actual example this was something meaningful. */
    private static bool boolean_condition(object o)
    {
        return true;
    }
}

File: issta2018artifacts/minimized-bugs/buggy-key/FourElementArray.java
// A simplified version of the bug fixed by Daikon commit d9e8fa8


public class FourElementArray {

    /* Every null in the function below was originally some meaningful String.
     * The original documentation specified that this function always returns
     * an array with four elements.
     */
        /*@ public normal_behavior
@ requires true;
@ ensures \result.length == 4;
@*/
    public static String[] esc_quantify(Object... vars) {
	if (vars.length == 1) {
	    //:: error: return.type.incompatible
	    return new String[] {null, null, ")"};
	} else if ((vars.length == 2) && boolean_condition(vars[1])) {
	    return new String[] {
		null,
		null,
		null,
		")"
	    };
	} else {
	    return new String[] {
		null, null, null, ")"
	    };
	}
    }

    /* In the actual example this was something meaningful. */
        /*@ public normal_behavior
@ requires true;
@ ensures true;
@*/
    private static boolean boolean_condition(Object o) {
	return true;
    }
}
